app/helpers/comments_helper.rb

```diff
@@ -22,6 +22,36 @@ def render_pings(comment, pingable: nil)
     end.html_safe
   end
 
+  def render_comment_helpers(comment_text, user = current_user)
```

Comment:
This is going to be a performance killer on a page with any more than a few of these helpers. Currently, this runs at least one query per helper usage. We should cache category lowercase-name/ID/human-name maps heavily - category names and IDs aren't going to change much, ever, so we can afford to basically permanently cache this and add a cache-breaker on category create/update.